test/controllers/SelfAssessmentControllerSpec.scala

```diff
@@ -170,4 +176,69 @@ class SelfAssessmentControllerSpec extends BaseSpec with CurrentTaxYear with Moc
       status(result) shouldBe BAD_REQUEST
     }
   }
+
+  "Calling SelfAssessmentController.viewPayments" should {
+
+    "return 200 and render viewPayments page" when {
+
+      "a user is an Activated SA user" in new LocalSetup {
+
+        override def fakeAuthJourney: FakeAuthJourney =
+          new FakeAuthJourney(ActivatedOnlineFilerSelfAssessmentUser(saUtr))
+
+        when(
+          mockSelfAssessmentPaymentsService.getPayments(any())(any(), any())
+        ) thenReturn Future.successful(List.empty)
+
+        val result: Future[Result] = controller.viewPayments()(FakeRequest())
+
+        status(result) shouldBe OK
+
+        Jsoup.parse(contentAsString(result)).text() should include(
```

Comment:
Dont need to do Jsoup parse here as contentAsString would still include message key if we are just checking whole source code contains this  string